src/compas/geometry/splits/splits.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from compas.utilities import pairwise
+
+from compas.datastructures import Mesh
+from compas.geometry import Polyline
+from compas.geometry import intersection_segment_plane
+from compas.geometry import intersection_mesh_plane
+from compas.geometry import subtract_vectors
+from compas.geometry import dot_vectors
+
+__all__ = [
+    'split_polyline_plane',
+    'split_mesh_plane',
+]
+
+
+def split_polyline_plane(polyline, plane):
+    """Split a polyline by a plane, returns a list of polylines, if there has been intersections
+    Returns list of the splitted polylines
+
+    Parameters
+    ----------
+    polyline : compas.datastructures.Polyline
+    plane : compas.geometry.Plane
+
+    Returns
+    -------
+    polylines : list of compas.geometry.Polyline
+    """
+    splitted_polylines_points = []
+    sublist = []
+    for i, segment in enumerate(pairwise(polyline)):
+        sublist.append(segment.start)
+        temp_intersection = intersection_segment_plane(segment, plane)
```

Comment:
the `temp_` part seems superfluous, it is just a local variable and its temporal status is clear by that already.